### PR TITLE
Automate census docker build

### PR DESCRIPTION
**Pull request by AdeHarrison**
# Motivation and Context
We want travis build to push docker images to Google Container Registry instead of dockerhub

# What has changed
Replaced docker image prefix with GCR registry location
NOTE: before merge, remove "automate-census-docker-build" branch condition and "branches only" section in .travis.yml file

# How to test?
From travis dashboard, switch to  "automate-census-docker-build" branch and "Restart Build". Once complete, check latest Docker image is in GCR.

# Links

[Trello](https://trello.com/c/dC6sEWqE/443-set-up-automated-docker-image-build)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,22 +18,18 @@ script:
...
+  docker push "${DOCKER_GCP_LOCATION}"/actionsvc;
+  fi
+- bash <(curl -s https://codecov.io/bash)
 
 notifications:
   slack:
     rooms:
       secure: ToyAMr3A3m53jITHwcGrH4rOxXfK6fIVyefbeeqeAL507zbOE+2HwRAphRv1DaYco9uu6B9w+5ao5AO3BO9BT3lgHxF00itKdnqhabZJVYsdRApeERoq625h2QyKHukDMrRV/Ml7m5lhVMOrvw0zNgjYBVJRL/K/VReCF6tTyHeiPSRepOWE+Ds5Qtu1GuaDxRMCljMy4eFlCJ/N/nIZSCWdOXMCVc2b8HXTqff3NNwXM9JVdUyibCdv1eydT9KIaDLlnBLEqGZwluKzJGg4CYZr6B4Yr/ciiGvofJEvZnrw5U2SJTqCgLPYHU/bHW6dJaB0ICpx94uMGSoyVTCNQDo1brBgP+w+9Z2IY0op9yOWghEpxAkNQL3FDC5BqprKbsDQG+yQpvXQ7XrlxxLp8ycNEuNTGsZRUwHhUzDsEousiAxA0bSJdS6LxSi11C9KDpaZH0cIhDEgktUtSlHNtPdX7GVRWXo4Nz9ZQuSG1Riobi2JX/k90AuQPOk9G5SVwBckp/st+pzYI02q3/FBEM+ZZSwb5WPalSOyGc20FotJL+59ruwhNOyOEX9/YrK9bEOHPTOMvYu43oGNzTc4VAmXva/OBf5Z5yP9VkTPwADtqljj7qbXcAMkPdLYAnyLhR4JC2vPuGcyxiX0uB+h45V1Uf0g/l5nLLWr4YvdPJc=
-    on_failure: always
+    on_failure: never
```
Comment (AdamHawtin) [inline .travis.yml]: Could this whole notifications section be deleted if you're turning them all off?

File: .travis.yml [hunk 2]
```diff
@@ -6,6 +6,11 @@ services:
 language: java
 jdk: openjdk8
 
+env:
+  global:
+  - SOURCE_IMAGE_NAME="actionsvc"
```
Comment (davidmort) [inline .travis.yml]: Is there any benefit the env vars given they're only used once?

File: .travis.yml [hunk 3]
```diff
@@ -18,28 +23,17 @@ script:
...
 
 cache:
   directories:
   - $HOME/.m2
 
 branches:
     only:
-        - master
\ No newline at end of file
+      - automate-census-docker-build
```
Comment (davidmort) [inline .travis.yml]: shouldn't this be "rm-census"?

File: .travis.yml [hunk 4]
```diff
@@ -18,28 +23,17 @@ script:
...
-    fi
-  - bash <(curl -s https://codecov.io/bash)
-
-notifications:
-  slack:
-    rooms:
-      secure: ToyAMr3A3m53jITHwcGrH4rOxXfK6fIVyefbeeqeAL507zbOE+2HwRAphRv1DaYco9uu6B9w+5ao5AO3BO9BT3lgHxF00itKdnqhabZJVYsdRApeERoq625h2QyKHukDMrRV/Ml7m5lhVMOrvw0zNgjYBVJRL/K/VReCF6tTyHeiPSRepOWE+Ds5Qtu1GuaDxRMCljMy4eFlCJ/N/nIZSCWdOXMCVc2b8HXTqff3NNwXM9JVdUyibCdv1eydT9KIaDLlnBLEqGZwluKzJGg4CYZr6B4Yr/ciiGvofJEvZnrw5U2SJTqCgLPYHU/bHW6dJaB0ICpx94uMGSoyVTCNQDo1brBgP+w+9Z2IY0op9yOWghEpxAkNQL3FDC5BqprKbsDQG+yQpvXQ7XrlxxLp8ycNEuNTGsZRUwHhUzDsEousiAxA0bSJdS6LxSi11C9KDpaZH0cIhDEgktUtSlHNtPdX7GVRWXo4Nz9ZQuSG1Riobi2JX/k90AuQPOk9G5SVwBckp/st+pzYI02q3/FBEM+ZZSwb5WPalSOyGc20FotJL+59ruwhNOyOEX9/YrK9bEOHPTOMvYu43oGNzTc4VAmXva/OBf5Z5yP9VkTPwADtqljj7qbXcAMkPdLYAnyLhR4JC2vPuGcyxiX0uB+h45V1Uf0g/l5nLLWr4YvdPJc=
-    on_failure: always
-    on_success: never
+- if ([ "$TRAVIS_BRANCH" == "rm-census" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]) || ([ "$TRAVIS_BRANCH" == "automate-census-docker-build" ]); then
```
Comment (davidmort) [inline .travis.yml]: need to remove the second part of this "if" as per the PR description before it can be merged

File: .travis.yml (hunk 3 above)
Comment (AdeHarrison) [reply .travis.yml]: fixed

**Pull request merged**